benchmarks/transformer/score_mod.py

```diff
@@ -1077,59 +1135,365 @@ def generate_experiment_configs(
                 backends=backends,
+                max_autotune=max_autotune,
             )
         )
 
     return all_configs
 
 
-def main(args):
+def _output_json_for_dashboard(
+    experiments,
+    output_file,
+    benchmark_name="PyTorch operator microbenchmark",
+):
+    """
+    Write the result into JSON format for PyTorch OSS dashboard.
+    The JSON format is defined at
+    https://github.com/pytorch/pytorch/wiki/How-to-integrate-with-PyTorch-OSS-benchmark-database
+
+    Args:
+        experiments: List of experiment results
+        output_file: Path to output JSON file
+        benchmark_name: Name of the benchmark
+    """
+    if not experiments:
+        return
+
+    import platform
+    from dataclasses import asdict, dataclass
+    from typing import Any, Optional
+
+    # Prepare headers and records for JSON output
+    records = []
+    for experiment in experiments:
+        config = experiment.config
+        results_dict = (
+            experiment.results
+        )  # This is a dict: backend -> ExperimentResults
+
+        # Process each backend result
+        for backend, results in results_dict.items():
+            # Extract data from experiment
+            test_name = f"{config.attn_type}_{backend}"
+            input_config = f"shape: {config.shape}, dtype: {config.dtype}"
+
+            # Determine mode based on backward pass
+            mode = "training" if config.calculate_bwd_time else "inference"
+
+            # Extract dtype
+            dtype = (
+                str(config.dtype).split(".")[1]
+                if "." in str(config.dtype)
+                else str(config.dtype)
+            )
+
+            # Determine device
+            device = "cuda"
+
+            # Get device architecture
+            device_arch = (
+                torch.cuda.get_device_name(0)
+                if device == "cuda"
+                else platform.processor()
+                if device == "cpu"
+                else "unknown"
+            )
+
+        # Create dataclasses for JSON structure
+        @dataclass
+        class BenchmarkInfo:
+            name: str
+            mode: Optional[str]
+            dtype: str
+            extra_info: dict[str, Any]
+
+        @dataclass
+        class ModelInfo:
+            name: str
+            type: str
+            origins: list[str]
+            extra_info: dict[str, Any]
+
+        @dataclass
+        class MetricInfo:
+            name: str
+            unit: str
+            benchmark_values: list[float]
+            target_value: Optional[float]
+
+        @dataclass
+        class BenchmarkRecord:
+            benchmark: BenchmarkInfo
+            model: ModelInfo
+            metric: MetricInfo
+
+        # Add record for forward latency
+        record_fwd_latency = BenchmarkRecord(
+            benchmark=BenchmarkInfo(
+                name=benchmark_name,
+                mode=mode,
+                dtype=dtype,
+                extra_info={
```

Comment:
this feels kinda verbose for what's going on - is there any way to reduce the copy / paste duplication throughout? e.g. we can at least precalculate `extra_info` once and use that for any records we need